### PR TITLE
unit tests for junos specific code

### PR DESCRIPTION
**Pull request by vnitinv**
### What does this PR do?
Few JUNOS specific bug fixes and unit test added

### What issues does this PR fix or reference?
Changes for using latest PyEZ

### Tests written?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by mirceaulinic** [comment]
Oh my... I personally find this hard to review: hundreds of changes in all the existing modules + tests in a single PR... I think you should split this into a PR with the corrections & a separate one just containing the tests.

**Comment by mirceaulinic** [comment]
Also: some rebase and squash would not hurt, it's very noisy and commits like https://github.com/saltstack/salt/pull/39884/commits/c67f68e0064151f9796ef07894bfb6235af00f79 don't tell me much.

**Comment by vnitinv** [comment]
@mirceaulinic We are consolidating all the changes (pull request from org members) into our org level branch and sending a unified pull request in a certain time (say each month once). This is as per discussion with SaltStack. In such scenario splitting feature specific pull request will be tough.

But at the same time, I take your second comment where a single blank line was added. Will pass on this feedback to the commiter. Thanks for pointing.

**Comment by cachedout** [comment]
@mirceaulinic FWIW, GitHub does give maintainers the option to squash PRs into a single commit when they are merged. So, I don't mind a lot of commits in PRs. Splitting features up though, as you mention, is very helpful to us. :]

**Comment by cachedout** [comment]
@vnitinv There are a few lint issues here. Could you please have a look?

**Comment by rajvidhimar** [comment]
@cachedout  I am unable to open https://jenkins.saltstack.com/ to check the error report. Is it down?

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
@rajvidhimar Yes, jenkins was having a rough day yesterday. It is back up now and I have restarted the test run. :)

**Review by cachedout — CHANGES_REQUESTED** [review]
Some lint changes are still needed here: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/9470/violations/file/tests/unit/modules/test_junos.py/

**Comment by vnitinv** [comment]
@cachedout with latest code we dont see any violation in this list
https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/9510/violations/

Is that something we are missing?

**Comment by rajvidhimar** [comment]
@cachedout https://jenkins.saltstack.com/job/PR/job/salt-pr-rs-cent7-n/9658/testReport/junit/unittest.loader.ModuleImportFailure.unit/modules/test_junos/
We have external dependencies such as nose, junos-eznc (PyEZ) to run the junos unit tests. Is there a way we can add these dependencies, so that the builds don't error out?

**Comment by vnitinv** [comment]
@cachedout Can you please review this and let us know if we need to do changes.

**Comment by cachedout** [comment]
Go Go Jenkins!

File: tests/unit/modules/test_junos.py [hunk 1]
```diff
@@ -0,0 +1,1679 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+__author__ = "Rajvi Dhimar"
+
+import unittest
+from nose.plugins.attrib import attr
```
Comment (s0undt3ch) [inline tests/unit/modules/test_junos.py]: We don't use nose and we're actually working towards using pytest.
`attr` seems just like a test selection decorator and, if that's the only benefit, could it be removed?

You can select this test case on the develop branch by passing the module filesystem path to runtests.py

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by cachedout** [comment]
@vnitinv We need this rebased because there is a merge conflict. This was reviewed by @s0undt3ch and he requested changes which are listed above. Thanks.

**Comment by vnitinv** [comment]
@cachedout I have done merge in place of rebase as rebase was giving me few errors. Hope merge should be fine. If you want we I send a new pull request with the cleaner version.

**Comment by vnitinv** [comment]
@cachedout @s0undt3ch Can you please review the changes and merge the pull request. 
All the jenkins jobs are passing.

**Review by cachedout — APPROVED** [review]

**Pull request merged**